src/lib.rs

```diff
@@ -518,6 +547,7 @@ where
 
         Ok(SortAndWriteHandler {
             cursor: 4096,
+            fadvise_cursor: 4096,
```

Comment:
This should NEVER be hard-coded.  Page sizes are not guaranteed to be 4k.  As I said you shouldn't be managing this manually at all, but if you feel that is necessary then you MUST query the kernel for the page size using `getpagesize`.  In most cases it will be 4kb, but if huge pages are involved then it could be substantially larger.  My comment to this effect was ignored on the previous mmap PR.  If you absolutely must use a static value, then at least use [`libc::_SC_PAGE_SIZE`](https://docs.rs/libc/0.2.84/libc/constant._SC_PAGE_SIZE.html) rather than a hard-coded magic number.